### ISSUE TITLE
Weaviate cannot handle tensor with different axis

### ISSUE THREAD

**Issue by samsja** [docarray/docarray#1600]
# Context

it seems that Weaviate cannot handle tensor with multiple axis (n-dimensional array)

If this is expected we should have a better error message and it should be documented somewhere

## How to reproduce.

weaviate version: 1.18.3
client version: 3.19.2

To run it please before start the weaviate database. (from the docarray repo)

```bash
 cd tests/index/weaviate
docker-compose up 
```

```python
from typing import Optional

from docarray.index import WeaviateDocumentIndex
from docarray.typing import ImageUrl, TorchTensor
from pydantic import Field

from docarray import BaseDoc, DocList
import torch

class NestedDoc(BaseDoc):
    embedding: Optional[TorchTensor] = Field(is_embedding=True)
    tensor: TorchTensor
    

class MyDoc(BaseDoc):
    product: NestedDoc

# Weviate Connection
batch_config = {
    "batch_size": 20,
    "dynamic": False,
    "timeout_retries": 3,
    "num_workers": 1,
}

runtimeconfig = WeaviateDocumentIndex.RuntimeConfig(batch_config=batch_config)

dbconfig = WeaviateDocumentIndex.DBConfig(host="http://localhost:8080")

store = WeaviateDocumentIndex[MyDoc](db_config=dbconfig)
store.configure(runtimeconfig)

docs = DocList[MyDoc]([MyDoc(product=NestedDoc(embedding=torch.zeros(128), tensor=torch.zeros(3,224,224)))])

store.index(docs)
```

```bash
{'error': [{'message': "invalid number array property 'product__tensor' on class 'MyDoc': invalid integer array value: [[[0.0 0
```

**Comment by samsja** [comment]
@hsm207 any idea ?

**Comment by hsm207** [comment]
@samsja this is expected. A list of list of list of numbers is not a supported [ data type](https://weaviate.io/developers/weaviate/config-refs/datatypes) in weaviate.

**Comment by samsja** [comment]
maybe we could have a better error message then ?

**Comment by hsm207** [comment]
@samsja what do you propose the error message should be?
